internal/metrics/metrics.go

```diff
@@ -14,3 +16,103 @@ func Register(serveMux *http.ServeMux) {
 		}),
 	))
 }
+
+const (
+	namespace           = "strong_duckling"
+	subSystemTcpChecker = "tcp_checker"
+)
+
+const (
+	strongswanVersion     = "strongswan_version"
+	strongDucklingVersion = "version"
+)
+
+type PrometheusReporter struct {
+	registry   prometheus.Registerer
+	version    *prometheus.GaugeVec
+	TcpChecker *TcpChecker
+}
+
+type TcpChecker struct {
```

Comment:
Could we make this non-exported? It is an implementation detail of the package and is not useful outside in it self?

Reply:
I guess so. Just tried to hide it behind a method that return the `tcpchecker.Reporter`, will push in a minute.